chromium_src/BUILD.gn

```diff
@@ -5,6 +5,9 @@
 
 source_set("browser") {
   sources = []
+  deps = [
+    "//brave/chromium_src/chrome/app"
```

Comment:
I think `//brave/chromium_src/chrome/app` should not be added here because it is higher layer than `chrome/browser`, `chrome/common`.
Instead, how about adding `//brave/chromium_src/chrome/app` to `//brave:browser_dependencies`? @jumde @bridiver 

When I needed to add new dependencies to `brave_main_delegate.cc`, I added them directlry to `//brave:browser_dependencies`. but I think I should move them into `//brave/chromium_src/chrome/app` later.